core/bifrost_test.go

```
@@ -895,6 +917,80 @@ func TestFilterProvidersByContext(t *testing.T) {
 	})
 }
 
+// TestListAllModels_SkipsCustomProvidersWithListModelsDisabled covers the fan-out
+// gate: a custom provider that disables list_models via allowed_requests must not
+// be dispatched at all, while the other providers still report their models. The
+// key lookup count is what separates "skipped" from "dispatched and bounced" —
+// both end up contributing no models to the aggregated response.
+func TestListAllModels_SkipsCustomProvidersWithListModelsDisabled(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprint(w, `{"object":"list","data":[{"id":"model-a","object":"model","created":0,"owned_by":"test"}]}`)
```

golangci-lint (2.13.2)

[error] Error return value of `fmt.Fprint` is not checked

(errcheck)